### PR TITLE
use timestamp based slot num to query blob

### PR DESCRIPTION
**Pull request by smtmfft**
Syncing https://github.com/taikoxyz/raiko/pull/76/files

### CODE REVIEW

**Comment by smtmfft** [comment]
Matched PI with taiko-a7.

File: lib/src/protocol_instance.rs [hunk 1]
```diff
@@ -109,7 +109,7 @@ pub fn assemble_protocol_instance(
         transition: Transition {
             parentHash: header.parent_hash,
             blockHash: header.hash(),
-            stateRoot: input.taiko.l1_header.state_root,
+            stateRoot: header.state_root,
```
Comment (smtmfft) [inline lib/src/protocol_instance.rs:112]: Is transition's stateRoot by design L1 or L2??

Comment (Brechtpd) [reply lib/src/protocol_instance.rs:112]: Oh yes it's the L2 one!

File: raiko-host/src/host/host.rs [hunk 2]
```diff
@@ -104,8 +104,15 @@ pub fn preflight(
         assert!(blob_hashs.len() >= 1);
         // Currently the protocol enforces the first blob hash to be used
         let blob_hash = blob_hashs[0];
+
+        let l2_chain_spec = get_network_spec(network);
+        let slot_id = block_time_to_block_slot(
+            l1_inclusion_block.header.timestamp.as_limbs()[0],
+            l2_chain_spec.genesis_time,
+            l2_chain_spec.seconds_per_slot,
+        )?;
```
Comment (Brechtpd) [inline raiko-host/src/host/host.rs:113]: Could something like `l1_inclusion_block_no - genesis_block_id` also work to avoid having to use timestamps?

Comment (smtmfft) [reply raiko-host/src/host/host.rs:113]: Any potential problem using timestamp??
I think using l1_inclusion_block_no is not work as it increases sequentially which is mismatch with the beacon slot id.

**Comment by CeciliaZ030** [comment]
@smtmfft Is A7 still under testing? Or you have stabilized the changes? 
@Brechtpd Should we separate these changes with flags and merge them?

**Comment by smtmfft** [comment]
> @smtmfft Is A7 still under testing? Or you have stabilized the changes? 
hmm, I think it is just stabilized enough for unstable use.

**Comment by Brechtpd** [comment]
lets just merge it in and support different networks like we already do, maybe in the futures using feature flags may make some tings easier but now it's just dynamic in the code which if it isn't extra work I think is pretty nice.

File: raiko-host/src/host/host.rs (hunk 2 above)
Comment (Brechtpd) [reply raiko-host/src/host/host.rs]: Just some extra dependencies on block times and time in general that you would not have with block ids

**Pull request merged**